src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js

```diff
@@ -261,280 +212,337 @@ class BasePaymentsPage extends React.Component {
-                        />
-                    </OfflineWithFeedback>
-                </View>
-                <AddPaymentMethodMenu
-                    isVisible={this.state.shouldShowAddPaymentMenu}
-                    onClose={this.hideAddPaymentMenu}
-                    anchorPosition={{
-                        horizontal: this.state.anchorPositionHorizontal,
-                        vertical: this.state.anchorPositionVertical - 10,
-                    }}
-                    onItemSelected={(method) => this.addPaymentMethodTypePressed(method)}
-                />
-                <Popover
-                    isVisible={this.state.shouldShowDefaultDeleteMenu}
-                    onClose={this.hideDefaultDeleteMenu}
-                    anchorPosition={{
-                        top: this.state.anchorPositionTop,
-                        right: this.state.anchorPositionRight,
-                    }}
+                    </View>
+                    {props.userWallet.currentBalance > 0 && (
+                        <View style={styles.mb3}>
+                            <KYCWall
+                                onSuccessfulKYC={navigateToTransferBalancePage}
+                                enablePaymentsRoute={ROUTES.SETTINGS_ENABLE_PAYMENTS}
+                                addBankAccountRoute={ROUTES.SETTINGS_ADD_BANK_ACCOUNT}
+                                addDebitCardRoute={ROUTES.SETTINGS_ADD_DEBIT_CARD}
+                                popoverPlacement="bottom"
+                            >
+                                {(triggerKYCFlow) => (
+                                    <MenuItem
+                                        title={translate('common.transferBalance')}
+                                        icon={Expensicons.Transfer}
+                                        onPress={triggerKYCFlow}
+                                        shouldShowRightIcon
+                                        disabled={props.network.isOffline}
+                                    />
+                                )}
+                            </KYCWall>
+                        </View>
+                    )}
+                </>
+            )}
+            <Text style={[styles.ph5, styles.textLabelSupporting, styles.mb1]}>{translate('paymentsPage.paymentMethodsTitle')}</Text>
+        </>
+    );
+
+    useEffect(() => {
+        PaymentMethods.openPaymentsPage();
+    }, []);
+
+    useEffect(() => {
+        // If the user was previously offline, skip debouncing showing the loader
+        if (!props.network.isOffline) {
+            updateShouldShowLoadingSpinner();
+        } else {
+            debounceSetShouldShowLoadingSpinner();
+        }
+    }, [props.network.isOffline, debounceSetShouldShowLoadingSpinner, updateShouldShowLoadingSpinner]);
+
+    useEffect(() => {
+        if (props.shouldListenForResize) {
+            setMenuPosition();
+        }
```

Comment:
This can be a separate effect

Reply:
Hi @Santhosh-Sellavel 
Alright, I simplified useEffect.